common/plugins/eu.esdihumboldt.hale.common.schema.persist/src/eu/esdihumboldt/hale/common/schema/persist/hsd/HaleSchemaUtil.java

```diff
@@ -33,4 +41,40 @@ public static InputStream getHaleSchemaXSD() {
 		return HaleSchemaUtil.class.getResourceAsStream("hsd.xsd");
 	}
 
+	/**
+	 * Combine the given schemas to a single schema.
+	 * 
+	 * @param schemas the schemas to combine
+	 * @param reporter the reporter if available
+	 * @return the combined schema
+	 */
+	public static Schema combineSchema(List<Schema> schemas, IOReporter reporter) {
+		if (schemas == null || schemas.isEmpty()) {
+			// empty schema
+			return new DefaultSchema(null, null);
+		}
+		else if (schemas.size() == 1) {
+			return schemas.get(0);
+		}
+		else {
+			DefaultSchema result = new DefaultSchema(null, null);
+
+			schemas.forEach(schema -> {
+				schema.getTypes().forEach(type -> {
+					if (reporter != null) {
+						// check if type is already there
+						QName name = type.getName();
+						if (result.getType(name) != null) {
+							reporter.warn(
+									MessageFormat.format("Multiple definitions of type {0}", name));
+						}
+					}
+					result.addType(type);
```

Comment:
I think the most likely situations here are:

1. There is no conflict (otherwise there was one in hale too, after loading the schemas, which we try to avoid for instance by allowing to load only one schema)
2. There is a conflict but the definitions are the same
3. There is a conflict and definitions differ

At that point we can't do much though I think. What would you think about raising the message from a warning to an error?